src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { ErrorMessages } from '../../types/ErrorMessages';
+import { createTodo } from '../../api/todos';
+
+interface Props {
+  isTodosPresent: boolean;
+  isAllTodosCompleted: boolean;
+  toggleAllTodos: () => void;
+  setTempTodo: (todo: Todo | null) => void;
+  setErrorMessage: (errorName: ErrorMessages) => void;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  USER_ID: number;
+}
+
+export const TodoHeader: React.FC<Props> = ({
+  isTodosPresent,
+  isAllTodosCompleted,
+  toggleAllTodos,
+  setTempTodo,
+  setErrorMessage,
+  setTodos,
+  USER_ID,
+}) => {
+  const [isAdding, setIsAdding] = useState<boolean>(false);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+
+  const handleNewTodoTitleChange
+  = (event: React.ChangeEvent<HTMLInputElement>) => (
+    setNewTodoTitle(event.target.value)
+  );
+
+  const clearNewTodoTitle = () => (
+    setNewTodoTitle('')
+  );
+
+  const addNewTodo = async (title: string) => {
+    if (!title) {
+      setErrorMessage(ErrorMessages.TitleError);
+
+      return;
+    }
+
+    try {
+      const newTodoPayload = {
+        completed: false,
+        title,
+        userId: USER_ID,
+      };
+
+      setTempTodo({
+        id: 0,
+        ...newTodoPayload,
+      });
+
+      const newTodo = await createTodo(newTodoPayload);
+
+      clearNewTodoTitle();
+      setTodos((prevState) => [...prevState, newTodo]);
+    } catch (error) {
+      setErrorMessage(ErrorMessages.TitleError);
+    } finally {
+      setTempTodo(null);
+    }
+  };
```

Comment:
Let's remove this function to the app component and provide it via props. 
It helps you remove unnecessary props in this component